File: backend/src/schemas/template.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TemplateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str
    category: str
    variant: str
    preview_url: str | None = None
    engine: str
    is_premium: bool
    is_ats_tested: bool
    created_at: datetime

File: frontend/src/types/template.ts
export type TemplateCategory =
  | "classic"
  | "modern"
  | "two-column"
  | "academic"
  | "cover-letter"
  | "creative";

export type TemplateEngine = "latex" | "typst";

export interface Template {
  readonly id: string;
  readonly name: string;
  readonly category: TemplateCategory;
  readonly variant: string; // e.g. "modern-blue", "classic-serif"
  readonly previewUrl: string;
  readonly pdfUrl?: string; // scrollable PDF shown in preview modal
  readonly engine: TemplateEngine;
  readonly isAtsTested: boolean;
  readonly createdAt: string;
}
